app/src/main/java/com/mth/example/buoi2telegram/message/MessageAdapter.kt

```diff
@@ -7,52 +7,50 @@ import android.widget.ImageView
 import android.widget.ProgressBar
 import android.widget.TextView
 import androidx.recyclerview.widget.DiffUtil
-import androidx.recyclerview.widget.RecyclerView.ViewHolder
 import androidx.recyclerview.widget.ListAdapter
-import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView.ViewHolder
 import com.mth.example.buoi2telegram.R
 
 enum class ItemType {
-    FRIENDMESS, MYMESS, LOADMORE
+    FRIEND_MESS, MY_MESS, LOAD_MORE
 }
 
-class MessageAdapter : ListAdapter<MessageItem, ViewHolder>(messageDiffUtil) {
-    private var isLoadingAdd: Boolean = false
+class MessageAdapter : ListAdapter<MessageItem, ViewHolder>(MessageDiffUtil) {
 
-    private var listData = mutableListOf<MessageItem>()
-
-    fun setData(list: MutableList<MessageItem>) {
-        this.listData = list
-        submitList(listData)
-    }
-
-    object messageDiffUtil : DiffUtil.ItemCallback<MessageItem>() {
+    object MessageDiffUtil : DiffUtil.ItemCallback<MessageItem>() {
         override fun areItemsTheSame(oldItem: MessageItem, newItem: MessageItem): Boolean {
             return oldItem::class.java.simpleName == newItem::class.java.simpleName
         }
 
         override fun areContentsTheSame(oldItem: MessageItem, newItem: MessageItem): Boolean {
-            return oldItem.id == newItem.id
+            return when (oldItem) {
```

Comment:
@TuanHiep20000705 đoạn này để diff xảy ra được (item mới khác item cũ thì mới submit list được) thì a đang fake cho content của msg là current time in millis.